### ISSUE TITLE
Update HTTP schema

### ISSUE THREAD

**Issue by MartinKolarik** [jsdelivr/globalping#171]
I missed this before since it wasn't documented, but all properties which are under `query` should be moved one level up, same as with DNS.

I'm also not very happy with the `path` name because the usual meaning of that word is [this](https://ldapwiki.com/images/url-example.png), but in our case, it is meant to contain the `query` part as well, which is very confusing. We already have a separate `port` field, so if we wanted to match that, we could add `query` too.

Another thing I find confusing is that with DNS type, the `port` is related to the `resolver`. Here we have the same fields, but `resolver` doesn't accept a custom port, and the `port` property is used to construct the URL.

I think this separation to multiple fields is not needed, and it may be better to remove all these fields and overload the `target` field for HTTP so that it contains the whole URL. E.g.:

```json
{
  "measurement":{
    "type":"http",
    "target":"google.com",
    "query":{
      "method":"GET",
      "path":"/npm/opipe",
      "port":80
    },
    "protocol":"HTTP",
    "resolver": "..."
  }
}
```

=> 

```json
{
  "type":"http",
  "target":"google.com:80/npm/opipe", // or http://google.com:80/npm/opipe
  "measurementOptions":{
    "method":"GET",
    "protocol":"HTTP",
    "resolver": "..."
  }
}
```

This seems more intuitive to me than breaking the URL into many parts and matches the philosophy from #159 that `measurementOptions` contains the less important "options" while most queries can be run without it. Also, it removes the confusion about what the `port` is related to.

One thing I'm not 100% sure about is whether we should also allow http/https in the target in this case, it would make it even closer to typical command line behavior which I is good.

**Comment by patrykcieszkowski** [comment]
The PR we pushed today moves `protocol`, `port`, and `resolver` outside the `query` object, and keeps `query` for request-specific values. Following `DNS` pattern. https://github.com/jsdelivr/globalping/blob/master/docs/measurement/schema/request.md#dns

As for the URL format, it may seem like a more ituitive option at first, but the fact you'd have to manually modify the string in order to test the server with different parameters cancells it out. Also, HTTP2 doesn't have its own prefix, so that complicates things a lot.

**Comment by MartinKolarik** [comment]
The PR changed `query` for DNS but not for HTTP as far as I can see. I didn't comment on the PR directly since HTTP wasn't in the original issue either.

> As for the URL format, it may seem like a more ituitive option at first, but the fact you'd have to manually modify the string in order to test the server with different parameters cancells it out. Also, HTTP2 doesn't have its own prefix, so that complicates things a lot.

What do you consider parameters here? I'm quite ok with keeping the protocol separate or allowing it in both the target and the options. But I see no reason for separating path + port + query, and there is a number of issues with that right now, as I described.

**Comment by patrykcieszkowski** [comment]
Yeah, it did.
https://github.com/jsdelivr/globalping/blob/master/src/measurement/schema/command-schema.ts#L25

**Comment by MartinKolarik** [comment]
Ok, it moved resolver and protocol but not the rest. I suppose my example was confusing as I copied it from the demo page, but I'm talking about method, host, path, headers.

**Comment by MartinKolarik** [comment]
There should be no `query` object at all. And if there is a `query` param, it should mean the query string. But I prefer the other version I described.

**Comment by patrykcieszkowski** [comment]
> What do you consider parameters here?

request-specific parameters. Anything a user might want to modify when querying e.g. an API - it's rather unlikely a server would run on multiple ports, protocols (other than redirections) and any actual query testing would involve changes in `path`, `host` and `headers`

**Comment by patrykcieszkowski** [comment]
> There should be no `query` object at all. And if there is a `query` param, it should mean the query string. But I prefer the other version I described.

ok, then why are we keeping `query` object for `DNS` requests?

**Comment by MartinKolarik** [comment]
> query

That's a good question. I suppose if we called it `request` here, it would match the DNS semantics 🤔

**Comment by patrykcieszkowski** [comment]
@jimaek what do you think?

**Comment by MartinKolarik** [comment]
Removing it is an option too but it's probably better to keep it wrapped, just with a better name. It gives the properties inside more clear meaning. E.g. for DNS, the `type` is a rather generic property name (we even have one level above) so nesting it in `query` makes it clear what it is related to.

**Comment by jimaek** [comment]
> > There should be no `query` object at all. And if there is a `query` param, it should mean the query string. But I prefer the other version I described.
> 
> ok, then why are we keeping `query` object for `DNS` requests?
> 
> I can see where the confusion might occur with it's naming. But for the sake of consistency - we gotta follow the same pattern everywhere. We should either remove it completely, or rename it.

I agree, we need to follow the same naming. I really don't see a problem with `query`, nobody will confuse it with query strings, especially since we dont have a query-strings dedicated parameter and everything is supposed to be in `path`.

Also please provide some json examples of the changes you want to make. Its a bit hard to follow due to the many changes.

`"target":"google.com:80/npm/opipe", ` this looks like a better UX but I wanted to keep the domain separate to keep the format consistent across measurements. 
So `target` would always be a valid ping/traceroute/mtr/dns input and then you use the additional HTTP-specific parameters for the rest. We could keep as it was and then rename `path` to something else to make it clearer that it contains everything including query-strings

**Comment by MartinKolarik** [comment]
Summary of discussions: no magic behavior for "target" (maybe later but probably not, and we let the tools handle this). Rename `query` object to `request` and separate `path` to `path` and `query` fields.

**Comment by MartinKolarik** [comment]
I think this was already covered by some PR, right @patrykcieszkowski ?

**Comment by patrykcieszkowski** [comment]
yup, its already merged

https://github.com/jsdelivr/globalping/pull/173
